Add blog post for 'Jenkins Design Library 3' (#7802)
jenkins-infra/jenkins.io: content/blog/2025/01/2025-01-10-design-library.adoc

---
layout: post
title: "Introducing Jenkins Design Library 3"
tags:
- jenkins
- user-experience
authors:
- janfaracik
opengraph:
  image: /images/post-images/2025/01/2025-01-10-design-library/opengraph.png
discourse: true
---

image::/images/post-images/2025/01/2025-01-10-design-library/opengraph.png[role=center]

> We’re thrilled to announce that link:https://weekly.ci.jenkins.io/design-library[**Jenkins Design Library 3**] is now live! 🎉

Reviewer: ```suggestion
We’re thrilled to announce that the link:https://weekly.ci.jenkins.io/design-library[**Jenkins Design Library 3**] is now live! 🎉
```

With this release, we’ve refined every aspect of the library to provide better documentation for plugin developers and ensure a more consistent,
accessible, and intuitive experience.

'''

=== So what's new?

**A new consistent design**

> Every page has been refined to maintain a cohesive look and feel, making it easier than ever to work with Jenkins
> Design Library. All samples are now grouped into two clear categories — Components and Patterns — ensuring you can quickly
> find what you need. We've also introduced interactive examples, such as the new link:https://weekly.ci.jenkins.io/design-library/spacing[Spacing] sample, which'll make it
> easier than ever to find the component or pattern best suited for your needs.

Reviewer: ```suggestion
Every page has been refined to maintain a cohesive look and feel, making it easier than ever to work with Jenkins Design Library.
All samples are now grouped into two clear categories — Components and Patterns — ensuring you can quickly find what you need.
We've also introduced interactive examples, such as the new link:https://weekly.ci.jenkins.io/design-library/spacing[Spacing] sample, which will make it easier than ever to find the component or pattern best suited for your needs.
```

**New ‘Edit this page’ button**

> Want to contribute or suggest improvements? The new 'Edit this page' button makes it easy to jump in and collaborate.

Reviewer: ```suggestion
Want to contribute or suggest improvements?
The new 'Edit this page' button makes it easy to jump in and collaborate.
```

**Accessibility improvements**

> The library now respects user preferences for contrast (prefers-contrast) and motion (prefers-reduced-motion), making it more accessible for everyone.

Reviewer: ```suggestion
The library now respects user preferences for contrast (prefers-contrast) and motion (prefers-reduced-motion), making it more accessible to everyone.
```

**Versions**

Pages can now declare which Long-Term Support (LTS) release they’re compatible with, so you always know where the latest components are available.

'''

=== Available now 🚀

> Jenkins Design Library 3 is ready for plugin developers to explore and use. Head over to link:https://weekly.ci.jenkins.io/design-library[Jenkins Design Library] to dive in and see all the updates in action.

Reviewer: ```suggestion
Jenkins Design Library 3 is ready for plugin developers to explore and use.
Head over to link:https://weekly.ci.jenkins.io/design-library[Jenkins Design Library] to dive in and see all the updates in action.
```

**Thanks to the community 🎉**

> A huge thank-you to everyone who contributed to this effort. Happy coding!

Reviewer: ```suggestion
A huge thank you to everyone who contributed to this effort.
Happy coding!
```
